x/migration/keeper/msg_server_claim_morse_application.go

```diff
@@ -137,12 +138,68 @@ func (k msgServer) ClaimMorseApplication(ctx context.Context, msg *migrationtype
 		morseClaimableAccount,
 	)
 
+	claimedAppStake := morseClaimableAccount.GetApplicationStake()
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+	sessionEndHeight := sharedtypes.GetSessionEndHeight(&sharedParams, sdkCtx.BlockHeight())
+	claimedUnstakedBalance := morseClaimableAccount.GetUnstakedBalance()
+
+	currentSessionStart := sharedtypes.GetSessionStartHeight(&sharedParams, sdkCtx.BlockHeight())
+	previousSessionEnd := sharedtypes.GetSessionEndHeight(&sharedParams, currentSessionStart-1)
+
 	// Query for any existing application stake prior to staking.
 	preClaimAppStake := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 0)
 	foundApp, isFound := k.appKeeper.GetApplication(ctx, shannonAccAddr.String())
 	if isFound {
 		preClaimAppStake = *foundApp.Stake
 	}
+	postClaimAppStake := preClaimAppStake.Add(morseClaimableAccount.GetApplicationStake())
+
+	// TODO_IN_THIS_COMMIT: comment...
+	// ... application began unbonding on Morse and the unbonding period has since elapsed...
+	// ... both stake and unstatked balance were minted to the shannonDestAddr account...
+	// ... no application was staked...
+	// - use a lookup table to get the est. block times per network...
+	// - use the shared params and the block time to calculate the unstake session end height...
+	// - emit an unbonding start event...
+	if !morseClaimableAccount.UnstakingCompletionTime.IsZero() {
+		unbondedApp := &apptypes.Application{
+			Address:                 shannonAccAddr.String(),
+			Stake:                   &postClaimAppStake,
+			UnstakeSessionEndHeight: uint64(previousSessionEnd),
+			// Services:             (intentionally omitted, no services were staked),
+			// ServiceConfigHistory: (intentionally omitted, no services were staked),
+		}
+
+		durationUntilUnstake := morseClaimableAccount.UnstakingCompletionTime.Sub(time.Now())
+		if durationUntilUnstake.Seconds() <= 0 {
+			// Emit an event which signals that the claimed Morse supplier's unbonding
+			// period began on Morse, and ended while waiting to be claimed.
+			morseSupplierUnbondingEndEvent := apptypes.EventApplicationUnbondingEnd{
+				Application:        unbondedApp,
+				Reason:             apptypes.ApplicationUnbondingReason_APPLICATION_UNBONDING_REASON_ELECTIVE,
```

Comment:
@Reviewier, there is no `APPLICATION_UNBONDING_REASON_UNSPECIFIED`. It could be added EITHER:
A) **Preserving backwards compatibility**: 
    - Using the next available enum index (not 0)
    - Against convention
B) **Following convention**:
    - Shifting the enum values to set UNSPECIFIED to 0
    - Not backwards compatible for existing consumers of migration protobuf types (assuming any exist)